packages/devextreme/testing/tests/DevExpress.ui.widgets/chat.tests.js

```diff
@@ -187,6 +189,165 @@ QUnit.module('Message group', moduleConfig, () => {
     });
 });
 
+QUnit.module('renderMessage', moduleConfig, () => {
+    QUnit.test('Chat items should be updated after renderMessage has been called', function(assert) {
+        const author = {
+            id: MOCK_CURRENT_USER_ID,
+        };
+
+        const newMessage = {
+            author,
+            timestamp: NOW,
+            text: 'NEW MESSAGE',
+        };
+
+        this.instance.renderMessage(newMessage, author);
+
+        const { items } = this.instance.option();
+        const lastItem = items[items.length - 1];
+
+        assert.strictEqual(lastItem, newMessage);
+    });
+
+    QUnit.test('Message List items should be updated after renderMessage has been called', function(assert) {
+        const author = {
+            id: MOCK_CURRENT_USER_ID,
+        };
+
+        const newMessage = {
+            author,
+            timestamp: NOW,
+            text: 'NEW MESSAGE',
+        };
+
+        this.instance.renderMessage(newMessage, author);
+
+        const { items } = this.instance._messageList.option();
```

Comment:
If we need test messageList we should test this component separately.

Reply:
I understand you. I try not to use the private options. But we are not writing separate tests for the internal components of Chat right now. However, at the same time, we need to test the functionality

Reply:
We discussed this earlier. You told, that in future development, independent tests for internal components will be created, and this test will be moved there. Okay, let's leave it as is.